control-plane/subcommand/server-acl-init/command.go

```diff
@@ -1032,6 +1055,25 @@ func (c *Command) quitVaultAgent() {
 	}
 }
 
+// serverIPAddresses attempts to refresh the server IPs using netaddrs methods. These 'raw' IPs are used
+// when boostrapping ACLs and before consul-server-connection-manager runs.
+func (c *Command) serverIPAddresses() ([]net.IPAddr, error) {
+	var ipAddrs []net.IPAddr
+	var err error
+	if err = backoff.Retry(func() error {
+		ipAddrs, err = netaddrs.IPAddrs(c.ctx, c.consulFlags.Addresses, c.log)
+		if err != nil {
+			c.log.Error("Error resolving IP Address", "err", err)
+			return err
+		}
+		c.log.Info("Refreshing server IP addresses", "addresses", ipAddrs)
+		return nil
+	}, exponentialBackoffWithMaxInterval()); err != nil {
+		return nil, err
+	}
+	return ipAddrs, nil
```

Comment:
Do we want the nested backoff/retry? There may be a case for resetting the parent loop, but it might be easier to reason about with one-level.

Reply:
I am not a fan of the style of it but we are using a similar retry style during the bootstrap (line 330) and I went with that for consistency.